sdk/daml-script/runner/src/upgrade-test/scala/com/daml/lf/engine/script/IdeLedgerRunnerUpgradeTest.scala

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.engine.script
+
+import com.daml.bazeltools.BazelRunfiles.{requiredResource, rlocation}
+import com.daml.lf.engine.script.test.DarUtil.{buildDar, Dar, DataDep}
+import com.daml.lf.language.LanguageVersion
+import com.daml.scalautil.Statement.discard
+import io.circe._
+import io.circe.yaml
+import java.io.File
+import java.nio.file.{Files, Path, Paths}
+import org.scalatest.wordspec.AsyncWordSpec
+import org.scalatest.Suite
+import scala.collection.mutable
+import scala.concurrent.Future
+import scala.jdk.CollectionConverters._
+import scala.sys.process.ProcessLogger
+import scala.util.matching.Regex
+
+final class IdeLedgerRunnerUpgradeTest extends AsyncWordSpec with RunnerTestBase {
+  self: Suite =>
+
+  val languageVersion: LanguageVersion = LanguageVersion.v1_dev
+
+  lazy val damlScriptDar = requiredResource("daml-script/daml3/daml3-script-1.dev.dar")
+  lazy val upgradeTestLibDar: Path = rlocation(Paths.get("daml-script/runner/upgrade-test-lib.dar"))
+
+  lazy val tempDir: Path = Files.createTempDirectory("ide-upgrade-test")
+
+  val testFileDir: Path = rlocation(Paths.get("daml-script/runner/src/upgrade-test/daml/test/"))
+  val testCases: Seq[TestCase] = getTestCases(testFileDir)
+
+  "daml-script upgrades on IDE Ledger" should {
+    testCases.foreach { testCase =>
+      testCase.name in {
+        for {
+          testDarPath <- buildTestCaseDar(testCase)
+          r <- testDamlScript(
+            testDarPath,
+            Seq(
+              "--ide-ledger",
+              s"--script-name=${testCase.name}:main",
+              "--enable-contract-upgrading",
```

Comment:
We'll omit this flag if Remy decides it should be inferred from script type + lf version